org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/debug/AbstractRunHTMLDebugTab.java

```diff
@@ -183,19 +269,21 @@ private void validateProgramPath() {
 					errorMessage = MessageFormat.format(
 							Messages.RunProgramTab_error_malformedUR, 
 							ex.getMessage());
+					urlDecoration.setDescriptionText(errorMessage);
+					urlDecoration.show();
+				}				
+				if(webRootText.getText().isBlank()) {
```

Comment:
> * Is any folder can be a Web-Root and a Workspace Folder  ;like  `${workspace_loc:/test-js}` is going to be accepted by debugger?

The debugger does not see `${workspace_loc:/test-js}`, because in `AbstractHTMLDebugDelegate` a string substitution is performed. The debugger will only see a plain path. 

> * If the Web-Root is an existing system path, why don't we check this?.. I see the only check for `blanc` path.  Anyway, if the value is manually typed in - it should be verified as it can be any erroneous text

Thanks for pointing that out. I added validation.